### PR TITLE
Tf project feature/tf project tests

### PR DESCRIPTION
**Pull request by TechnicallyWilliams**
## What is the current behavior?
-------------------------------------

Currently there are no working examples of how to generate a mock interface for testability of the AzDev clients.

## What is the new behavior?
-------------------------------------

Code base now has a working example of a mocked interface being injected into a method that news up the CoreClient using GoMock to assert that calls are being made in an expected way.

**The following are links to gomock that were used as a reference for building the testing pattern:**
https://github.com/golang/mock
https://godoc.org/github.com/golang/mock/gomock#Call.MinTimes

**Installation:**
go get github.com/golang/mock/gomock
go install github.com/golang/mock/mockgen

**Command used to generate mock interfaces:**
"mockgen -source=foo.go -destination=mock_client_file.go"

## New Issues?
-------------------------------------
- Current unit tests are living within the main package/namespace but should be in their own package/namespace. This invites us to rethink the overall project packaging structure and may require a project packaging structure that more directly conforms to go conventions. #29 
- Need a script that auto stubs methods that define an interface for a client using some combination of "grep" and "awk" #30

### CODE REVIEW

File: terraform-provider/provider-src/src/client.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
+package main
```
Comment (nmiodice) [inline terraform-provider/provider-src/src/client.go]: Do we need this? I think the file is different in master, it should go there and not in a new client.go file if possible.

File: terraform-provider/provider-src/src/resource_project.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package main
```
Comment (nmiodice) [inline terraform-provider/provider-src/src/resource_project.go]: File looks out of date

File: terraform-provider/provider-src/src/resource_project_test.go [hunk 3]
```diff
@@ -37,7 +22,13 @@ func TestProjectCreate_MapsTfState(t *testing.T) {
...
\ No newline at end of file
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	coreClient := NewMockCoreClient(ctrl)
+	coreClient.EXPECT().QueueCreateProject(gomock.Not(gomock.Nil()), gomock.Not(gomock.Nil())).Return(nil, errors.New("Whoops")).MinTimes(1).MaxTimes(1)
+	clients := &AggregatedClient{CoreClient: coreClient}
+	
+	_, err := projectCreate(context.Background(), clients, testValues)
+	require.Equal(t, "Whoops", err.Error())
```
Comment (nmiodice) [inline terraform-provider/provider-src/src/resource_project_test.go]: This is AWESOME!!

File: terraform-provider/provider-src/src/client.go (hunk 1 above)
Comment (TechnicallyWilliams) [reply terraform-provider/provider-src/src/client.go]: This file was deleted. Reference to package main for newing up clients lives in azdo_config.go

File: terraform-provider/provider-src/src/resource_project.go (hunk 2 above)
Comment (TechnicallyWilliams) [reply terraform-provider/provider-src/src/resource_project.go]: Method calls in this file were updated to match their definitions which no longer live in client.go but instead now live in azdo_config.go

Comment (awkwardindustries) [reply terraform-provider/provider-src/src/resource_project.go]: Reverted to master version.

**Pull request merged**